app/models/detailed_guide.rb

```diff
@@ -76,8 +76,26 @@ def translatable?
     true
   end
 
+  def related_mainstream_base_path
+    url = related_mainstream_content_url
+    parse_base_path_from_related_mainstream_url(url)
+  end
+
+  def additional_related_mainstream_base_path
+    url = additional_related_mainstream_content_url
+    parse_base_path_from_related_mainstream_url(url)
+  end
+
   private
 
+  def parse_base_path_from_related_mainstream_url(url)
+    return nil if url.nil? || url.empty?
+    parsed_url = URI.parse(url)
+    url_is_invalid = !parsed_url.host.include?('gov.uk')
```

Comment:
This should only be things that are hosted on GOV.UK itself. This would consider URLs on e.g. https://www.ofgem.gov.uk/ valid.